include/boost/json/impl/object.ipp

```diff
@@ -822,4 +823,30 @@ destroy(
 
 BOOST_JSON_NS_END
 
+//----------------------------------------------------------
+//
+// std::hash specialization
+//
+//----------------------------------------------------------
+
+namespace std {
+std::size_t 
+hash<::boost::json::object>::operator()(
+    ::boost::json::object const& jo) const noexcept 
+{
+    std::size_t seed = jo.size();
+    for (const auto& kv_pair : jo) {
+        const auto hk = std::hash<boost::json::string>{}(kv_pair.key());
```

Comment:
This constructs a temporary `json::string` from `key()` (which is a `string_view`). I don't think that's what you intended :)

Reply:
Unfortunately std::hash<std::string_view> is C++17. :-)

Reply:
Use
```
        auto const hk = ::boost::json::detail::digest(
            kv_pair.key().data(), kv_pair.key().size(), 0);
``